### PR TITLE
Miserable bugfix and Ionic support

### PR DESCRIPTION
**Pull request by SergeyMell**
Fixed issue with trimming `null` value
Added Ionic elements support

### CODE REVIEW

File: projects/ngx-trim-directive/src/lib/ngx-trim.directive.ts [hunk 1]
```diff
@@ -149,7 +151,7 @@ export class NgxTrimDirective implements OnInit, OnDestroy {
     if ((this.trim === '' || 'blur' === this.trim) && 'function' === typeof value.trim && value.trim() !== value) {
 
       NgxTrimDirective.trimValue(el, value);
-      NgxTrimDirective.dispatchEvent(el, 'blur'); // in case updateOn is set to blur
+      NgxTrimDirective.dispatchEvent(el, event.type); // in case updateOn is set to blur
```
Comment (KingMario) [inline projects/ngx-trim-directive/src/lib/ngx-trim.directive.ts:154]: This is exact the value `blue` set for updateOn, a coincidence with the event.type of this handler.

File: projects/ngx-trim-directive/src/lib/ngx-trim.directive.ts [hunk 2]
```diff
@@ -32,9 +32,10 @@ export class NgxTrimDirective implements OnInit, OnDestroy {
 
     const elem = this.elementRef.nativeElement;
     const eleValue = elem.value;
-    if (trimOption !== false && eleValue !== eleValue.trim()) {
+    if (trimOption !== false && eleValue && eleValue !== eleValue.trim()) {
       // initially trim the value if needed
       NgxTrimDirective.dispatchEvent(elem, 'blur');
+      NgxTrimDirective.dispatchEvent(elem, 'ionBlur');
```
Comment (KingMario) [inline projects/ngx-trim-directive/src/lib/ngx-trim.directive.ts:38]: Should ionBlur also be handled?

**Comment by KingMario** [comment]
Thanks, @SergeyMell.

Because I'm not family with Ionic, could you please confirm that if `ionBlur` event should also be handled.

Thanks for the pull request and your contribution making this library better.

**Comment by SergeyMell** [comment]
Yes, you can find it here, for example
https://ionicframework.com/docs/api/input#events

Actually, supporting Ionic could be not very important (just required for my project), but also I've fixed one place where you're trimming the value which could also be `null` and this is causing an error. I could remove the Ionic support and leave only this fix

**Review by KingMario — COMMENTED** [review]
> Yes, you can find it here, for example
> https://ionicframework.com/docs/api/input#events
> 
if so, the line https://github.com/KingMario/packages/pull/16/files#diff-8d66416fb5042baf003211c87b18ea07R138 should also be updated?

> Actually, supporting Ionic could be not very important (just required for my project), but also I've fixed one place where you're trimming the value which could also be `null` and this is causing an error. I could remove the Ionic support and leave only this fix

**Comment by KingMario** [comment]
@SergeyMell Could you please reproduce the trim-of-null issue using stackblitz? I think the value of an input element could not be null.

And for ionic, I tried to copy the directive to an ionic application and found it doesn't trim the value on input or blur. I guess more modification is needed.

**Comment by KingMario** [comment]
@SergeyMell I have tested the modification with Ionic and it doesn't work in Ionic. Hence temporarily close this PR.

Thanks a lot.

**Pull request closed without merging**